pandas/io/parquet.py

```diff
@@ -53,9 +53,15 @@ def validate_dataframe(df: DataFrame):
         if not isinstance(df, DataFrame):
             raise ValueError("to_parquet only supports IO with DataFrames")
 
-        # must have value column names (strings only)
-        if df.columns.inferred_type not in {"string", "empty"}:
-            raise ValueError("parquet must have string column names")
+        # must have value column names for all index levels (strings only)
+        if isinstance(df.columns, MultiIndex):
+            if not all(
```

Comment:
can you add a test for this

Reply:
@jreback Since the enhancement is to add support for MultiIndex columns in parquet and the check is to handle MultiIndex columns in addition to single-level Index columns, I've added both round-trip and error-on-write tests for both MultiIndexes and single-level Indexes for columns in parquet format.

Do let me know if the coverage from the added tests are sufficient based on my latest commit - and whether additional tests are needed. :)